pkg/controllers/state/node.go

```diff
@@ -53,17 +62,49 @@ func (c *NodeController) Reconcile(ctx context.Context, req reconcile.Request) (
 		}
 		return reconcile.Result{}, err
 	}
+
 	if err := c.cluster.updateNode(ctx, node); err != nil {
 		return reconcile.Result{}, err
 	}
-	// ensure it's aware of any nodes we discover, this is a no-op if the node is already known to our cluster state
+
+	if err := c.record(ctx, node); err != nil {
+		logging.FromContext(ctx).Errorf("Failed to update gauges: %s", err)
+		return reconcile.Result{}, err
+	}
+
 	return reconcile.Result{Requeue: true, RequeueAfter: stateRetryPeriod}, nil
 }
 
+// TODO: Determine if additional watches conflict w/ cluster state
 func (c *NodeController) Register(ctx context.Context, m manager.Manager) error {
 	return controllerruntime.
 		NewControllerManagedBy(m).
 		Named(nodeControllerName).
 		For(&v1.Node{}).
+		Watches(
```

Comment:
Yeah, we don't want to do this.  When the node is updated we do a full poll of the nodes pods which let us resync our state if we are off (which hopefully doesn't occur). This occurs about once a minute when kubelet updates the node.  Performing that action every time a pod on the node changes won't scale well.